lib/AgrirouterClient/src/AgrirouterClient.cpp

```diff
@@ -15,59 +16,79 @@ AgrirouterClient::AgrirouterClient()
     // HTTP is default ConnectionProvider
     Settings *settings = new Settings();
     settings->setConnectionType(Settings::HTTP);
-    this->m_chunkSize = DEFAULT_CHUNK_SIZE;
+    m_chunkSize = DEFAULT_CHUNK_SIZE;
     this->init(settings);
 }
 
 AgrirouterClient::AgrirouterClient(Settings *settings)
 {
-    this->m_chunkSize = DEFAULT_CHUNK_SIZE;
+    m_chunkSize = DEFAULT_CHUNK_SIZE;
     this->init(settings);
 }
 
 AgrirouterClient::AgrirouterClient(Settings *settings, uint32_t chunkSize)
 {
-    this->m_chunkSize = chunkSize;
+    m_chunkSize = chunkSize;
     this->init(settings);
 }
 
 void AgrirouterClient::init(Settings *settings)
 {
     m_seqNo = 0;
     m_settings = settings;
-    m_messageProvider = new MessageProvider(settings, this->m_chunkSize);
+    m_messageProvider = new MessageProvider(settings, m_chunkSize);
```

Comment:
This can cause a memory leak.
Check if m_messageProvider is NULL first. If it is not, delete the existing object before creating a new one.
Also, init m_messageProvider in header file or initializer list with NULL.